parl/algorithms/torch/sac.py

```diff
@@ -0,0 +1,149 @@
+#   Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import parl
+import torch
+from torch.distributions import Normal
+import torch.nn.functional as F
+from copy import deepcopy
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+__all__ = ['SAC']
+
+
+class SAC(parl.Algorithm):
+    def __init__(self,
+                 model,
+                 max_action,
+                 discount=None,
+                 tau=None,
+                 alpha=None,
+                 actor_lr=None,
+                 critic_lr=None,
+                 policy_freq=1,
+                 automatic_entropy_tuning=False,
+                 entropy_lr=None,
+                 action_dim=None):
+        """ SAC algorithm
+            Args:
+                model(parl.Model): forward network of actor and critic.
+                max_action (float): the largest value that an action can be, env.action_space.high[0]
+                discount(float): discounted factor for reward computation
+                tau (float): decay coefficient when updating the weights of self.target_model with self.model
+                alpha (float): Temperature parameter determines the relative importance of the entropy against the reward
+                actor_lr (float): learning rate of the actor model
+                critic_lr (float): learning rate of the critic model
+                policy_freq(int): frequency to train actor(& adjust alpha if necessary) and update params
+                automatic_entropy_tuning(bool): whether or not to adjust alpha automatically
+                entropy_lr: learning rate of entropy
+                action_dim: the dimension of an action, env.action_space.shape[0], to calculate target_entropy
+        """
+        assert isinstance(discount, float)
+        assert isinstance(tau, float)
+        assert isinstance(alpha, float)
+        assert isinstance(actor_lr, float)
+        assert isinstance(critic_lr, float)
+        assert isinstance(entropy_lr, float)
```

Comment:
Is `entropy_lr` a necessary parameter?  if not, please remove this assert line.

Reply:
done